app/src/main/java/com/github/ybecker/epforuml/CoursesFragment.kt

```diff
@@ -96,29 +73,70 @@ class CoursesFragment : Fragment() {
+                }
 
         }
 
         override fun getItemCount() = courses.size
 
-
     }
 
     private inner class CourseViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
         private val courseTitleTextView = itemView.findViewById<TextView>(R.id.courseTitleTextView)
         private val subscriptionSwitch = itemView.findViewById<Switch>(R.id.subscriptionSwitch)
+        private val notificationSwitch = itemView.findViewById<Switch>(R.id.notificationSwitch)
 
         fun bind(course: Course) {
             courseTitleTextView.text = course.courseName
             subscriptionSwitch.isEnabled = user.userId.length != 0
-            subscriptionSwitch.isChecked = userSubscriptions.map { it.courseId }.contains(course.courseId)
+            val is_subscribed = userSubscriptions.map { it.courseId }.contains(course.courseId)
+            subscriptionSwitch.isChecked = is_subscribed
+            notificationSwitch.isEnabled = is_subscribed
+            db.getCourseNotificationUserIds(course.courseId).thenAccept {
+
+                notificationSwitch.isChecked = it.contains(user.userId)
+
+                // add a listener to every switches
+                subscriptionSwitch.setOnCheckedChangeListener { _, isChecked ->
+                    if(course != null){
+                        if (isChecked) {
+                            // add a subscription for this user in the course
+                            db.addSubscription(user.userId, course.courseId)
+                            notificationSwitch.isEnabled = true
+                            Toast.makeText(itemView.context,"You subscribed to " + course.courseName,Toast.LENGTH_SHORT).show()
+                        }
+                        else {
+                            db.removeSubscription(user.userId, course.courseId)
+                            db.removeNotification(user.userId, course.courseId)
+                            notificationSwitch.isEnabled = false
+                            notificationSwitch.isChecked = false
+                            Toast.makeText(itemView.context,"You unsubscribed to " + course.courseName,Toast.LENGTH_SHORT).show()
+
+                        }
+                    }
+                }
+                notificationSwitch.setOnCheckedChangeListener {_, isChecked ->
+                    if(course != null){
+                        if (isChecked) {
+                            // add a subscription for this user in the course
+                            db.addNotification(user.userId, course.courseId)
+                            Toast.makeText(itemView.context, "You allow notifications for " + course.courseName, Toast.LENGTH_SHORT).show()
+                        }
+                        else {
+                            db.removeNotification(user.userId, course.courseId)
+                            Toast.makeText(itemView.context,"You reject notifications for " + course.courseName,Toast.LENGTH_SHORT).show()
```

Comment:
Maybe we could add a method to create and show Toasts as we are using it at several places